scripts/specializations/CpAIWorker.lua

```diff
@@ -341,7 +339,12 @@ function CpAIWorker:stopCurrentAIJob(superFunc, message, ...)
                 end
             end
         end
+        local job = self:getJob()
+        if not job:isFinishingAllowed(message) then 
```

Comment:
Oh, so the idea is intercept the stopCurrentAIJob() from the implements when empty? Can't we handle that within the strategy?